core/base/src/main/java/org/eclipse/dataspaceconnector/core/defaults/DefaultServicesExtension.java

```diff
@@ -0,0 +1,96 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core.defaults;
+
+import org.eclipse.dataspaceconnector.common.concurrency.LockManager;
+import org.eclipse.dataspaceconnector.core.defaults.assetindex.InMemoryAssetIndex;
+import org.eclipse.dataspaceconnector.core.defaults.contractdefinition.InMemoryContractDefinitionStore;
+import org.eclipse.dataspaceconnector.core.defaults.negotiationstore.InMemoryContractNegotiationStore;
+import org.eclipse.dataspaceconnector.core.defaults.policystore.InMemoryPolicyStore;
+import org.eclipse.dataspaceconnector.core.defaults.transferprocessstore.InMemoryTransferProcessStore;
+import org.eclipse.dataspaceconnector.dataloading.AssetLoader;
+import org.eclipse.dataspaceconnector.dataloading.ContractDefinitionLoader;
+import org.eclipse.dataspaceconnector.spi.asset.AssetIndex;
+import org.eclipse.dataspaceconnector.spi.asset.DataAddressResolver;
+import org.eclipse.dataspaceconnector.spi.contract.negotiation.store.ContractNegotiationStore;
+import org.eclipse.dataspaceconnector.spi.contract.offer.store.ContractDefinitionStore;
+import org.eclipse.dataspaceconnector.spi.policy.store.PolicyStore;
+import org.eclipse.dataspaceconnector.spi.system.Provider;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.transfer.store.TransferProcessStore;
+
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Provides (in-mem) defaults for various stores, registries etc.
+ * Provider methods are only invoked if no other implementation was found on the classpath.
+ */
+public class DefaultServicesExtension implements ServiceExtension {
+
+    private static InMemoryAssetIndex assetIndex;
+    private final InMemoryContractDefinitionStore contractDefinitionStore;
+
+    public DefaultServicesExtension() {
+        assetIndex = new InMemoryAssetIndex();
+        contractDefinitionStore = new InMemoryContractDefinitionStore();
```

Comment:
also the contract definition store should be loaded lazily, and the initialization in the constructor avoided